### PR TITLE
Code Insight: Adds focus management integration tests for line and pie charts

### PR DESCRIPTION
**Pull request by vovakulikov**
Closes https://github.com/sourcegraph/sourcegraph/issues/42312

## Background
This adds two test cases for pie and line charts where we test their focusable elements (title, context menu, chart parts UI like points or pie units, ...etc.). This PR doesn't include a bar chart this is because the focus is broken in the bar chart at the moment. We will solve this problem in a separate issue and add a focus test case for the bar chart right after it. 

## Test plan
- Make sure that newly added test passes 
- Review new aria-labels in the insight card layouts 

## App preview:

- [Web](https://sg-web-vk-add-focus-mngt-integration-test.onrender.com)
- [Storybook](https://5f0f381c0e50750022dc6bf7-dhlqmzqbvd.chromatic.com)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.

### CODE REVIEW

File: client/web/src/integration/insights/insight/insight-chart-focus.test.ts [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+            assert.strictEqual(
+                await hasFocus(driver, `[aria-label="Pie chart"] a:nth-child(${arcIndex + 1})`),
+                true,
+                'Insight pie arc should be focused'
+            )
+        }
+    })
+})
+
+async function hasFocus(driver: Driver, selector: string): Promise<boolean> {
```
Comment (valerybugakov) [inline client/web/src/integration/insights/insight/insight-chart-focus.test.ts:132]: What do you think about moving it to `client/shared/src/testing`?

File: client/wildcard/src/components/Charts/components/pie-chart/PieChart.tsx [hunk 2]
```diff
@@ -53,7 +52,10 @@ export function PieChart<Datum>(props: PieChartProps<Datum>): ReactElement | nul
     // hovered arc last in arcs array. By that we sort of change z-index (ordering) of svg element
     // and put hovered label and arc over other arc elements
     const [hoveredArc, setHoveredArc] = useState<PieArcDatum<Datum> | null>(null)
-    const sortedData = useMemo(() => distributePieArcs(data, getDatumValue), [data, getDatumValue])
+    const sortedData = useMemo(() => [...data].sort((first, second) => getDatumValue(second) - getDatumValue(first)), [
```
Comment (valerybugakov) [inline client/wildcard/src/components/Charts/components/pie-chart/PieChart.tsx:55]: What's the reason for this change?

**Review by valerybugakov — APPROVED** [review]

File: client/wildcard/src/components/Charts/components/pie-chart/PieChart.tsx (hunk 2 above)
Comment (vovakulikov) [reply client/wildcard/src/components/Charts/components/pie-chart/PieChart.tsx:55]: `distributePieArcs` was an original logic for shuffling pie arcs [small, big, small, big, ....], we've done this because it wasn't easy to aim small arcs on the chart. Since we have a dynamic z-index logic in the PieChart we can remove `distributePieArcs` sorting.
